src/main/java/org/opensearch/neuralsearch/processor/util/ProcessorUtils.java

```diff
@@ -77,6 +78,9 @@ public static boolean validateRerankCriteria(
      */
     public static float getScoreFromSourceMap(final Map<String, Object> sourceAsMap, final String targetField) {
         Object val = getValueFromSource(sourceAsMap, targetField).get();
+        if (val instanceof String) {
+            return Float.parseFloat((String) val);
+        }
         return ((Number) val).floatValue();
```

Comment:
nit: we need to throw exception if the object is neither String nor Number, can be done later